man/fi_poll.3.md

```diff
@@ -159,9 +170,10 @@ struct fi_wait_attr {
 
 - *FI_WAIT_FD*
 : Indicates that the wait set should use a file descriptor as its wait
-  mechanism.  A file descriptor wait object must be usable in select,
-  poll, and epoll routines.  However, a provider may signal an FD wait
-  object by marking it as readable, writable, or with an error.
+  mechanism. Since it may not always be possible to use a single underlying
+  wait object, the file descriptor is not guaranteed to be usable in select,
+  poll, and epoll routines. However, when possible, a provider may signal an FD
+  wait object by marking it as readable, writable, or with an error.
```

Comment:
No - if the provider is using epoll, it should only need to return the epoll fd.  This would be for the case where epoll couldn't be used, or isn't for some performance reason.  I'm just trying to think of how to handle this in a generic fashion.